### ISSUE TITLE
Ability to play audio disappeared

### ISSUE THREAD

**Issue by davidanton** [davidsansome/tsurukame#512]
Sometime during the day today (August 1) the ability to play audio disappeared from the app. The play icons disappeared and no audio plays when it's configured to play automatically.
Some users have reported the same issue on the [WaniKani forums](https://community.wanikani.com/t/ios-tsurukame-native-app-with-offline-lessons-and-reviews/30635/894) 
I tried reinstalling, restarting my iPhone.
iPhone X running iOS 14.7.1

**Comment by UInt2048** [comment]
I can confirm that today, **September** 1st, this issue occurs upon a full refresh with WaniKani, whether automatic or manual.

**Comment by UInt2048** [comment]
Issue identified: WaniKani has shifted to a format such as https://files.wanikani.com/28dym1qiib6mcbojjxfophlqlklw which is a webm file. Thus, there are no audio/mpeg files for this code to run:

https://github.com/davidsansome/tsurukame/blob/db521e8142dbb38f40dbfd3dba55d52098591cb1/ios/WaniKaniAPI/Sources/WaniKaniAPI/Subject.swift#L210-L216

**Comment by vietqhoang** [comment]
Please see my explanation in [the community post](https://community.wanikani.com/t/ios-tsurukame-native-app-with-offline-lessons-and-reviews/30635/899?u=viet)

TL;DR: We did not replace the mp3 format with webm. We offer three audio formats: mp3, ogg, and webm. We did change the hostname of the cdn URL and the file name pattern.

Looking at the code I believe it is dying on lines 212 and 213

https://github.com/davidsansome/tsurukame/blob/db521e8142dbb38f40dbfd3dba55d52098591cb1/ios/WaniKaniAPI/Sources/WaniKaniAPI/Subject.swift#L212-L213

Specifically, on line 212, the code is relying on the old filename pattern being hyphen separated. The new filenames don't follow this pattern, which is probably breaking line 213.

Old url:  https://cdn.wanikani.com/audios/3020-subject-2467.mp3?1547862356
New url: https://files.wanikani.com/28dym1qiib6mcbojjxfophlqlklw

**Comment by UInt2048** [comment]
@vietqhoang Thank you for the clarification - I didn't realize that the formats were still offered since there were no file extensions.

**Comment by davidsansome** [comment]
Thanks for reporting this! I'm taking a look now!

**Comment by davidsansome** [comment]
I've fixed the bug and I'm pushing an update to the app store now - it should be available in a couple of days.

Offline audio is currently broken but I can fix that later - I'll need to build and upload new audio packages. Anybody that had previously downloaded offline audio packages will need to delete them and download them again.

**Comment by davidsansome** [comment]
I'll leave this issue open until the update is released to help people find it.

**Comment by joeio** [comment]
I got the update on my phone but still no audio. Does it need to be removed and reinstalled completely?

If it does, will I only loose the app settings?

Thanks!

**Comment by contextnerror** [comment]
So I downloaded Tsurukame for the first time while this bug was still in effect. It looks like the update has gone out, but even with automatic audio for correct answers I’ve still never heard anything. Can someone tell me when audio is supposed to play so I can check if it’s working?

**Comment by UInt2048** [comment]
@contextnerror Audio is only available for vocabulary items. To test if audio is working at all, find one and there should be a little audio play icon next to the reading. ;)

**Comment by UInt2048** [comment]
@davidsansome I think it's safe to close - the update has been out for a while.

**Comment by UInt2048** [comment]
Update: Audio plays correctly, but an audio icon shows for kanji even though there's not actually audio for them.

**Comment by UInt2048** [comment]
Some forum users are still reporting issues.

**Comment by davidsansome** [comment]
Offline audio should be fixed in the next release by 20f5c8c014cb3e3bca2b2b04a0edacdd09240fd9
